src/App.tsx

```diff
@@ -14,11 +16,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const goodsWithId = goodsFromServer.map((good) => {
+  return {
+    name: good,
+    id: uniqid(),
+  };
+});
+
+interface State {
+  selectedGoods: string[]
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  reset = () => (
+    this.setState({ selectedGoods: [] })
+  );
+
+  addItem = (item: string) => {
+    const { selectedGoods } = this.state;
+
+    return this.setState({ selectedGoods: [...selectedGoods, item] });
```

Comment:
use function as argument in the setState
```suggestion
    return this.setState((prevState) => [
      ...prevState.selectedGoods,
      item,
    ]);
```
![image](https://user-images.githubusercontent.com/36257947/131633749-27faa515-be9c-45f6-92fa-15017e62adac.png)